File: services/backend/src/main/kotlin/edu/pwr/backend/config/DataSourceContextHolder.java
//package edu.pwr.backend.config;
//
//import java.util.ArrayList;
//import java.util.List;
//
//public class DataSourceContextHolder {
//    private static final ThreadLocal<String> threadLoal = new ThreadLocal<>();
//    private static List<String> datSourceLookupKeys = new ArrayList<>();
//
//    public static void switchDataSource(String ds) {
//      threadLoal.set(ds);
//
//    }
//
//    public static String getDataSource() {
//        return threadLoal.get();
//    }
//
//    public static void clearDataSource() {
//        threadLoal.remove();
//    }
//
//    public static boolean containDataSource(String ds) {
//        return datSourceLookupKeys.contains(ds);
//    }
//
//
//    public static void addDataSource(String ds) {
//        if (!containDataSource(ds)) {
//            datSourceLookupKeys.add(ds);
//        }
//    }
//
//
//
//
//
//}

File: services/backend/src/main/kotlin/edu/pwr/backend/config/DataSourceContextHolder.kt
package edu.pwr.backend.config

object DataSourceContextHolder {
    private val threadLocal = ThreadLocal<String>()
    private val datSourceLookupKeys: MutableList<String> = ArrayList()

    fun switchDataSource(ds: String) {
        threadLocal.set(ds)
    }

    fun getDataSource(): String {
        return threadLocal.get() ?: throw IllegalStateException("No data source set.")
    }


    fun clearDataSource() {
        threadLocal.remove()
    }

    fun containDataSource(ds: String): Boolean {
        return datSourceLookupKeys.contains(ds)
    }


    fun addDataSource(ds: String) {
        if (!containDataSource(ds)) {
            datSourceLookupKeys.add(ds)
        }
    }
}